process/smartContract/process.go

```diff
@@ -193,12 +193,15 @@ func (sc *scProcessor) GasScheduleChange(gasSchedule map[string]map[string]uint6
 	defer sc.mutGasLock.Unlock()
 
 	apiCosts := gasSchedule[core.ElrondAPICost]
-	if apiCosts == nil {
+	_, existAsyncCallStepField := apiCosts[core.AsyncCallStepField]
+	_, existAsyncCallbackGasLockField := apiCosts[core.AsyncCallbackGasLockField]
+	if apiCosts == nil || !existAsyncCallStepField || !existAsyncCallbackGasLockField {
 		return
 	}
 
 	builtInFuncCost := gasSchedule[core.BuiltInCost]
-	if builtInFuncCost == nil {
+	_, existsESDTTransfer := builtInFuncCost[core.BuiltInFunctionESDTTransfer]
+	if builtInFuncCost == nil || !existsESDTTransfer {
```

Comment:
Same as above. First check if builtInFuncCost is nil

Reply:
:+1: fixed